python/taichi/lang/ast/ast_transformer.py

```diff
@@ -1157,6 +1177,23 @@ def build_static_for(ctx, node, is_grouped):
         else:
             build_stmt(ctx, node.iter)
             targets = ASTTransformer.get_for_loop_targets(node)
+            iter_time = 0
+            if ti_unroll_limit:
+                for target_values in node.iter.ptr:
+                    iter_time += 1
+                    if iter_time > ti_unroll_limit:
+                        warnings.warn_explicit(
+                            f"loop unrolling only support under {ti_unroll_limit}",
+                            SyntaxWarning,
+                            ctx.file,
+                            node.lineno + ctx.lineno_offset,
+                            module="taichi",
+                        )
+
+                        node.iter.func = node.iter.args[0].func
+                        node.iter.args = node.iter.args[0].args
+                        return ASTTransformer.build_For(ctx, node)
```

Comment:
Don't fall back to normal loop. You can advice them to use normal loop, but you shouldn't force them.

Reply:
you mean just mention it?  not do fallback auto, right?

Reply:
yes.